dmutils/filters.py

```diff
@@ -24,18 +25,19 @@ def format_links(text):
                                 \.                        # match dot in host name
                                 (?:[^\s/?\.#<>"']+\.?)+   # match rest of domain
                                 (?:/[^\s<>"']*)?          # rest of url
-                                [^\s<>"'\.]               # no dot at end
+                                [^\s<>,"'\.]              # no dot at end
                                 )""", re.X)
-    matches = url_match.findall(text)
-    link = '{0}<a href={1} rel="external">{1}</a>'
-    if matches:
-        textArray = url_match.split(text)
+    matched_urls = url_match.findall(text)
+    if matched_urls:
+        link = '{0}<a href="{1}" rel="external">{2}</a>'
         formattedText = ""
-        for text in textArray:
-            if text in matches:
-                formattedText = link.format(formattedText, Markup.escape(text))
+        textArray = url_match.split(text)
+        for partial_text in textArray:
+            if partial_text in matched_urls:
+                url = "http://{}".format(partial_text) if partial_text.startswith('www') else partial_text
+                formattedText = link.format(formattedText, Markup.escape(url), Markup.escape(partial_text))
             else:
-                formattedText = '{}{}'.format(formattedText, Markup.escape(text))
+                formattedText = '{}{}'.format(formattedText, Markup.escape(partial_text))
```

Comment:
Since python strings are immutable, concatenating parts one by one leads to a lot of copying. It's usually better to gather parts into an array and then `u''.join(fomatted_text_parts)` in the end